libwasmvm/src/api.rs

```diff
@@ -312,6 +312,19 @@ impl BackendApi for GoApi {
         };
         gas_info.cost += callee_instance.create_gas_report().used_internally;
 
+        // copy callee event manager's info to caller instance
+        if !callee_instance.env.is_storage_readonly() {
```

Comment:
Why does the callee instance only copy to the caller when it has read-write permission?

Reply:
This is because events can be issues by instances having write permission.